internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2096,8 +2096,10 @@ func resourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{})
 		d.Set("public_network_access_enabled", publicNetworkAccess != managedclusters.PublicNetworkAccessDisabled)
 
 		upgradeChannel := ""
-		if profile := props.AutoUpgradeProfile; profile != nil && *profile.UpgradeChannel != managedclusters.UpgradeChannelNone {
-			upgradeChannel = string(*profile.UpgradeChannel)
+		if profile := props.AutoUpgradeProfile; profile != nil {
+			if v := profile.UpgradeChannel; v != nil {
+				upgradeChannel = string(*profile.UpgradeChannel)
+			}
 		}
```

Comment:
this means we'll be setting the value `None` into the state instead of an empty string, so I think this wants to be:

```suggestion
		if profile := props.AutoUpgradeProfile; profile != nil && profile.UpgradeChannel != nil && *profile.UpgradeChannel != managedclusters.UpgradeChannelNone {
			upgradeChannel = string(*profile.UpgradeChannel)
		}
```